mono/metadata/icall.c

```diff
@@ -6236,6 +6237,11 @@ ves_icall_System_Delegate_CreateDelegate_internal (MonoReflectionType *type, Mon
 		}
 	}
 
+	if (sig->generic_param_count && method->wrapper_type == MONO_WRAPPER_NONE) {
+		if (!method->is_inflated)
+			mono_raise_exception(mono_get_exception_argument("method", " Cannot bind to the target method because its signature differs from that of the delegate type"));
```

Comment:
Please use `mono_set_pending_exception` and return instead of using `mono_raise_exception`.